lib/modules/platform/scm.ts

```diff
@@ -0,0 +1,15 @@
+import { DefaultGitScm } from './default-scm';
+import { GithubScm } from './github/scm';
+import type { PlatformScm } from './types';
+
+export const platformScmImpls = new Map<string, PlatformScm>();
+//here you can register additional custom implementations..
+platformScmImpls.set('github', GithubScm.instance);
```

Comment:
```suggestion
platformScmImpls.set('github', GithubScm);
```
only store the constructor 
also set for all supported platforms.

Reply:
you mean like this:

```javascript
export const platformScmImpls = new Map<PlatformId, Constructor<PlatformScm>>();
platformScmImpls.set('azure', DefaultGitScm);
platformScmImpls.set('codecommit', DefaultGitScm);
platformScmImpls.set('bitbucket', DefaultGitScm);
platformScmImpls.set('bitbucket-server', DefaultGitScm);
platformScmImpls.set('gitea', DefaultGitScm);
platformScmImpls.set('github', GithubScm);
platformScmImpls.set('gitlab', DefaultGitScm);
```

no default scm-instance?

Reply:
yes

Reply:
@viceice with this new approach (scm-proxy + no-default-scm implementation) I've much problems with many tests.

i have to change every mock of git.$methodX to scm.$methodX in every test.....

Reply:
I think you need to set default scm on global test init. like we do for mocking platform globally. checkout test/setup.ts